tests/functional/u/used/used_before_assignment_typing.py

```diff
@@ -7,6 +7,15 @@
 if TYPE_CHECKING:
     if True:  # pylint: disable=using-constant-test
         import math
+    elif input():
+        import pickle
+    else:
+        with input() as temp:
+            import posix
+    if TYPE_CHECKING:
+        pass
+    else:
+        import math
```

Comment:
Nothing fails if I comment this out.

Reply:
My intention of this was to ensure that `math` is not imported by line 18, and `used-before-assignment` will be raised.

Reply:
Shouldn't `used-before-assignment` _not_ be raised if `math` is imported at runtime? I'm still having trouble connecting this highlighted portion to the assertions, but I'm hoping you can help me out.

Reply:
Correct me if I am wrong, but my understanding is that `math` is not imported at runtime as lines 15-18 are within a TYPE_CHECKING block (refer to line 7). Therefore, assertions will be raised for any usages of `math`.

Reply:
Thanks, that helps. I see now this is here to prevent a bad implementation of the proposed change that treats `import math` under an else as not guarded by TYPE_CHECKING because here it is guarded two levels up, and this case would have already passed because we were not excluding anything under else branches at all 👍